testsuite/shared/src/main/java/org/jboss/as/test/layers/LayersTest.java

```diff
@@ -71,31 +71,68 @@ public class LayersTest {
      * @param unused The set of modules that are OK to not be provisioned when all layers are provisioned.
      * If more modules than this set are not provisioned, it means that we are missing some modules and
      * an error occurs.
-     * @throws Exception
+     * @throws Exception on failure
      */
+    @Deprecated
     public static void test(String root, Set<String> unreferenced, Set<String> unused) throws Exception {
+        testBootInstallations(root);
+        testDeployedModules(root, unreferenced, unused);
+    }
+
+    /**
+     * Scan and check all available boot sequences booted successfully for an installation.
+     * @param root Path to installation
+     * @throws Exception on failure
+     */
+    public static void testBootInstallations(String root) throws Exception {
         File[] installations = new File(root).listFiles(File::isDirectory);
-        Result reference = null;
-        Result allLayers = null;
-        Map<String, Result> layers = new TreeMap<>();
         ExecutorService executor = Executors.newSingleThreadExecutor();
         try {
             for (File f : installations) {
                 Path installation = f.toPath();
-                Result res = Scanner.scan(installation, getConf(installation));
-                if (f.getName().equals(REFERENCE)) {
-                    reference = res;
-                } else if (f.getName().equals(ALL_LAYERS)) {
+                if (f.getName().equals(ALL_LAYERS)) {
                     checkExecution(executor, installation);
-                    allLayers = res;
                 } else {
                     checkExecution(executor, installation);
-                    layers.put(f.getName(), res);
                 }
             }
         } finally {
             executor.shutdownNow();
         }
+    }
+
+    /**
+     * Checks only allowed modules were loaded with the installation
```

Comment:
I'd say

"Checks the expected modules were provisioned with the @{code test-standalone-reference} and @{test-all-layers} installations."